backend/init/init.go

```diff
@@ -38,6 +39,7 @@ func init() {
 		"consul": func() backend.Backend { return backendconsul.New() },
 		"inmem":  func() backend.Backend { return backendinmem.New() },
 		"s3":     func() backend.Backend { return backendS3.New() },
+		"azure":  func() backend.Backend { return backendAzure.New() },
```

Comment:
thinking out loud: would it be worth naming this `azurerm` to match the provider?